src/Component/retrieveAll/PartnerRetriever.php

```diff
@@ -17,10 +17,15 @@ public function __construct(PartnerRepository $repository)
     }
 
     /**
+     * @param string $filter
      * @return array
      */
-    public function getAll(): array
+    public function getAll(?string $filter)
     {
+        if (!empty($filter)) {
```

Comment:
ici pour simpliier un petit peu je ferais ca 

$params = ($filter) ? ['firstName' => strtoupper($filter)] : ['id' => 'DESC'];

return $this->repo->findBy(['isActive' => true], $params) ?? [];  // ici ton double point d'intterro devrait reprendre la reponse si elle existe sinon envoyer par default un array vide.

Reply:
ok merci